### PR TITLE
[CI] Add GitHub Actions

### PR DESCRIPTION
**Pull request by Margen67**
Example run: https://github.com/Margen67/xenia/actions/runs/106069905

This PR adds automatic Windows and Linux builds, and Linting via GitHub Actions, and adds a badge to README.md.

### Pros:
  * Builds multiple configurations at a time. Therefore, the Debug configuration can be added without additional building time.

### Cons:
  * Artifacts:
      * Doesn't have a direct link to latest artifacts.
        This is possible with releases, but is messy.
      * Can't use ~~wildcards~~/regex, or upload files without zipping them.
        As a consequence, checked builds cannot be passworded without being double-zipped.
        This is also why multiple robocopy commands have to be done instead of one 7z command.
  * Can't delete builds/runs.

---

tl;dr Add this because AppVeyor is slow.

### CODE REVIEW

**Comment by DrChat** [comment]
Nice work :)
It might be worth investigating if we should start releasing builds of Xenia through the releases page. We could tag releases or just release every time a new commit is pushed to the master branch.

**Comment by Margen67** [comment]
@DrChat That's what Canary does, but it leads to a mess of tags: https://github.com/xenia-canary/xenia-canary/releases

**Comment by q4a** [comment]
Hi all. I like idea of using GitHub Actions. Is there any problem with merging this PR?

**Comment by q4a** [comment]
Nobody will add GitHub Actions?
